### ISSUE TITLE
Is it possible to start system programmatically from a Java main method?

### ISSUE THREAD

**Issue by mwnorman** [kumuluz/kumuluzee#84]
Similar to Wildlfy-Swarm ...

public static void main(String[] args) throws Exception {
    int port = 9090;
    Container container = new Container("MyFavouriteContainerName", port);
    container
      .dataSource("MyDS", (ds) -> {
        ds.driverName("h2");
        ds.connectionUrl("jdbc:h2:mem:test;DB_CLOSE_DELAY=-1;DB_CLOSE_ON_EXIT=FALSE");
        ds.userName("sa");
        ds.password("sa");
      })
    .deploy("name of Resource", MyResource.class, options)
    ;
    container.start();

**Comment by TFaga** [comment]
Hey,

Thanks for the heads up. Yes it is possible. The entire config can be programatically build and passed to the main `EeApplication` class.

Although there is currently an issue with this. Certain parameters defaults in the config are not correctly set, which means you currently need to manually set them. It'll be fixed in the next minor update that will include the necessary and sensible defaults for these values.

Current minimum example with a datasource (the server parts will be optional in the next update):

```java
public static void main(String args[]) {

        ServerConnectorConfig.Builder httpBuilder = new ServerConnectorConfig.Builder();
        httpBuilder.port(8080);

        ServerConnectorConfig.Builder httpsBuilder = new ServerConnectorConfig.Builder();
        httpsBuilder.port(8443);
        httpsBuilder.keystorePath("");
        httpsBuilder.keystorePassword("");

        ServerConfig.Builder serverBuilder = new ServerConfig.Builder();
        serverBuilder.http(httpBuilder);
        serverBuilder.https(httpsBuilder);

        DataSourceConfig.Builder dsBuilder = new DataSourceConfig.Builder();
        dsBuilder.jndiName("jdbc/H2DS");
        dsBuilder.connectionUrl("jdbc:h2:mem:test;DB_CLOSE_DELAY=-1;DB_CLOSE_ON_EXIT=FALSE");
        dsBuilder.username("sa");
        dsBuilder.password("sa");

        EeConfig.Builder eeConfigBuilder = new EeConfig.Builder();
        eeConfigBuilder.server(serverBuilder);
        eeConfigBuilder.datasource(dsBuilder);

        new EeApplication(eeConfigBuilder.build());
 }
```

**Comment by TFaga** [comment]
Fixed in [`2.5.2`](https://github.com/kumuluz/kumuluzee/releases/tag/v2.5.2)

You can now specified just the values you need to change;

```java
public static void main(String args[]) {

        DataSourceConfig.Builder dsBuilder = new DataSourceConfig.Builder();
        dsBuilder.jndiName("jdbc/H2DS");
        dsBuilder.connectionUrl("jdbc:h2:mem:test;DB_CLOSE_DELAY=-1;DB_CLOSE_ON_EXIT=FALSE");
        dsBuilder.username("sa");
        dsBuilder.password("sa");

        EeConfig.Builder eeConfigBuilder = new EeConfig.Builder();
        eeConfigBuilder.datasource(dsBuilder);

        new EeApplication(eeConfigBuilder.build());
    }
```

**Comment by wmacibnc** [comment]
Prezados,

Antes funciona o requisição para um resource que configurei, agora não está funcionando mais. Teria alguma configuração extra?

``
@RequestScoped
@Path("/v1/filtro")
@Produces(MediaType.APPLICATION_JSON)
public class FiltroResource {

	@GET
	@Path("teste")
	public Response teste() {
		Map<String, String> json = new HashMap<String, String>();
		json.put("teste", "teste");
		System.out.println(json);
		return Response.status(Response.Status.OK).entity(json).build();
	}
}
